llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -525,9 +525,10 @@ void CompileTimePropertiesPass::parseAlignmentAndApply(
   // Get the global variable with the annotation string.
   const GlobalVariable *AnnotStrArgGV = nullptr;
   const Value *IntrAnnotStringArg = IntrInst->getArgOperand(1);
-  if (auto *GEP = dyn_cast<GEPOperator>(IntrAnnotStringArg))
-    if (auto *C = dyn_cast<Constant>(GEP->getOperand(0)))
-      AnnotStrArgGV = dyn_cast<GlobalVariable>(C);
+  if (IntrAnnotStringArg->getType()->isOpaquePointerTy())
+    AnnotStrArgGV = dyn_cast<GlobalVariable>(IntrAnnotStringArg);
+  else if (auto *GEP = dyn_cast<GEPOperator>(IntrAnnotStringArg))
```

Comment:
The handling for this case (which does not seem to be related to opaque pointers) has changed.  What's the rationale for this change?

Reply:
This is for the opaque pointer. If the arg on ptr.annotation is an opaque pointer we do things differently